osx arm64 fixes

dslarm opened the pull request:
<!--
Thank you for your pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

regro-cf-autotick-bot committed:
migration: OSXArm

regro-cf-autotick-bot committed:
MNT: Re-rendered with conda-build 25.3.2, conda-smithy 3.47.1, and conda-forge-pinning 2025.04.04.19.35.36

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found some lint.

Here's what I've got...


For **recipe/meta.yaml**:

 * ❌ This recipe is using a compiler, which now requires adding a build dependence on `{{ stdlib("c") }}` as well. Note that this rule applies to each output of the recipe using a compiler. For further details, please see https://github.com/conda-forge/conda-forge.github.io/issues/2102.

For **recipe/meta.yaml**:

 * ℹ️ License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

Documentation on acceptable licenses can be found [here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions ).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14360728784](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14360728784). Examine the logs at this URL for more detail.</sub>

dslarm committed:
Update meta.yaml: fix linter errs

dslarm committed:
Update meta.yaml: add perl-packagers

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipe/meta.yaml**:

 * ℹ️ License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

Documentation on acceptable licenses can be found [here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions ).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14553395938](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14553395938). Examine the logs at this URL for more detail.</sub>

dslarm committed:
Update conda-forge.yml: cross-compile due to lack of travis

dslarm commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 25.3.2, conda-smithy 3.47.1, and conda-forge-pinning 2025.04.09.17.30.33

dslarm committed:
Update meta.yaml: try expat in build (osx-arm64 fails on missing expat)

dslarm committed:
Update meta.yaml: revert expat addn - didn't make a difference

dslarm committed:
Update build.sh: add debug

dslarm committed:
Update build.sh: debug

dslarm committed:
Update build.sh: debug

kynan commented:
@dslarm Are you intending for this to supersede #12 ?

dslarm commented:
> @dslarm Are you intending for this to supersede #12 ?

Hi @kynan - sort of.   As a regular user, I can't modify #12, I can only fork that PR from the bot, and extend it. There's nothing clever in my PR, just more failure :(.   I may not have time to look this week or next - so if you can try to fix, just use any PR :)

kynan commented:
@dslarm Both #12 and #13 are stuck on the same failure 🙃

dslarm committed:
test to see what is happening with compiler in check-devel

dslarm committed:
Merge pull request #1 from dslarm/main

test to see what is happening with compiler in check-devel

dslarm committed:
amend patch

dslarm committed:
amend patch

dslarm committed:
amend patch

dslarm committed:
amend patch

dslarm committed:
amend patch

dslarm committed:
amend patch

dslarm committed:
amend patch

dslarm committed:
amend patch

dslarm committed:
amend patch

dslarm committed:
amend patch

dslarm committed:
amend patch

dslarm committed:
amend patch

dslarm committed:
amend patch

dslarm committed:
- remove previous patch / debugging attempts
- add patch to fix test for libexpat to not run a target binary if
  cross compiling

dslarm commented:
@conda-forge-admin, please rerender

dslarm commented:
This should be ready to go.
The issue was a test for libexpat tried to execute a binary on the build host - but that binary was built for the target, not the build host - as we still cross-compile for apple silicon.

dslarm commented:
@kynan - it may be worth update maintainers to conda-forge/perl-packagers to share the load.

Commit:
MNT: Re-rendered with conda-build 25.4.1, conda-smithy 3.47.1, and conda-forge-pinning 2025.04.19.00.08.06
